src/content/manager/package.manager.ts

```diff
@@ -91,26 +100,43 @@ export class PackageManager extends BaseManager {
     }
 
     private buildPushUrl(): string {
-        let pushUrl = `${PackageManager.BASE_URL}/${PackageManager.IMPORT_ENDPOINT_PATH}`;
-        if (this.newKey) {
-            pushUrl = `${pushUrl}?newKey=${this.newKey}`;
-        }
-        if (this.overwrite) {
-            if (this.newKey) {
-                logger.error(
-                    "You cannot overwrite a package and set a new key at the same time. Please use only one of the options."
-                );
-                process.exit();
-            } else {
-                pushUrl = `${pushUrl}?overwrite=${this.overwrite}`;
-            }
-        }
-        return pushUrl;
+        this.validateOptions();
```

Comment:
IMO, this is not the place to validate the options, since the method's responsibility is to build the push url. I think such validation things should be handled in `package.command.ts`, before the factories are even created.

Reply:
I'm don't have a strong opinion about it either. It just feels more natural having it here, as if we store that validation logic inside of the `package.command.ts` it looks a bit odd IMO, as I think that file should have the responsibility of converting the commands to actual logic calls which do the work for them, otherwise I don't think we should have that at all. Ideally we should have some validator file maybe that does that throughout the app 🤔